src/main/java/org/broadinstitute/hellbender/tools/spark/sv/StringKmerizer.java

```diff
@@ -0,0 +1,76 @@
+package org.broadinstitute.hellbender.tools.spark.sv;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * Iterator over successive Kmers from a String.
+ * Silently skips over parts of the sequence that has characters other than A, C, G, or T.
+ *
+ * Created by tsharpe on 12/10/15.
+ */
+public final class StringKmerizer implements Iterator<Kmer>
```

Comment:
It's more efficient this way, taking advantage of an array access vs. a method call and avoiding a costly copy operation, but in the spirit of compromise I've rewritten it as one class, Kmerizer, that operates on a CharSequence rather than a String.  And I've written a shim class, ASCIICharSequence, that allows you to treat a byte[] as a CharSequence.  An overloaded constructor and toKmer method uses the shim to boost a byte[] into a CharSequence.